File: Data Structures & Algorithms/sort-colors/submission-0.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        
        int l=0;
        int r=nums.size()-1;
        int m=0;
        while(m<=r){
            
            if(nums[m]==0){
                int temp=nums[l];
                nums[l]=nums[m];
                nums[m]=temp;
                l++;
                m++;

            }else if(nums[m]==1){
                m++;
            }else{
                int temp=nums[r];
                nums[r]=nums[m];
                nums[m]=temp;
                r--;
            }
        }
        
    }
};

File: Data Structures & Algorithms/rotate-array/submission-0.java
class Solution {
    public void rotate(int[] nums, int k) {
        int l=k-1;
        int r=nums.length-1;
        while(l>=0){
            int temp=nums[l];
            nums[l]=nums[r];
            nums[r]=temp;
            r--;
            l--;
        }
        
    }
}